pkg/operation/botanist/component/etcd/etcd_copy_backups_task.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcd
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/extensions"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CopyInterface interface {
+	component.DeployWaiter
+	SetBackupConfig(druidv1alpha1.StoreSpec)
+	SetSourceBackupConfig(druidv1alpha1.StoreSpec)
+}
+
+type CopyBackupsTaskValues struct {
+	Name         string
+	Namespace    string
+	TargetStore  druidv1alpha1.StoreSpec
+	SourceStore  druidv1alpha1.StoreSpec
+	MaxBackups   *uint32
+	MaxBackupAge *uint32
+}
+
+type etcdCopyBackupsTask struct {
+	values       *CopyBackupsTaskValues
+	logger       logrus.FieldLogger
+	client       client.Client
+	waitInterval time.Duration
+	waitTimeout  time.Duration
+
+	etcdCopyBackupsTask *druidv1alpha1.EtcdCopyBackupsTask
+}
+
+func NewEtcdCopyBackupsTask(
+	logger logrus.FieldLogger,
+	client client.Client,
+	values *CopyBackupsTaskValues,
+	waitInterval time.Duration,
+	waitTimeout time.Duration) CopyInterface {
+
+	return &etcdCopyBackupsTask{
+		values,
+		logger,
+		client,
+		waitInterval,
+		waitTimeout,
+		&druidv1alpha1.EtcdCopyBackupsTask{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      values.Name,
+				Namespace: values.Namespace,
+			},
+		},
+	}
+}
+
+func (e *etcdCopyBackupsTask) Deploy(ctx context.Context) error {
+	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, e.client, e.etcdCopyBackupsTask, func() error {
+		metav1.SetMetaDataAnnotation(&e.etcdCopyBackupsTask.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile)
+		metav1.SetMetaDataAnnotation(&e.etcdCopyBackupsTask.ObjectMeta, v1beta1constants.GardenerTimestamp, TimeNow().UTC().String())
```

Comment:
These 2 annotations are not needed:
* The `operation=reconcile` annotation does not apply to `EtcdCopyBackupsTask` resources (it will not cause a reconcile).
* The `timestamp` annotation will not be useful if the resource is being recreated.